File: cmd/whoami.go
package cmd

import (
	"fmt"

	"github.com/mturley/agent-handler/db"
	"github.com/mturley/agent-handler/discover"
	"github.com/spf13/cobra"
)

var whoamiCmd = &cobra.Command{
	Use:   "whoami",
	Short: "Print the current session ID",
	RunE: func(cmd *cobra.Command, args []string) error {
		sessionID, err := discover.ResolveSessionID(db.HandlerHome())
		if err != nil {
			return fmt.Errorf("no session registered for this process. Run 'handler register' or start a new prompt first")
		}

		fmt.Print(sessionID)
		return nil
	},
}

func init() {
	whoamiCmd.GroupID = "agent"
	rootCmd.AddCommand(whoamiCmd)
}

File: src/Handler/Cmd/Whoami.hs
-- | Port of cmd/whoami.go: print the current session ID.
module Handler.Cmd.Whoami (whoamiCommand) where

import qualified Data.Text.IO as TIO
import Options.Applicative

import Handler.Cli.Common
import Handler.Db (handlerHome)
import Handler.Discover (resolveSessionId)

whoamiCommand :: Mod CommandFields NamedCommand
whoamiCommand = mkCommand "whoami" "Print the current session ID" (pure runWhoami)

runWhoami :: Ctx -> IO ()
runWhoami _ = do
  home <- handlerHome
  resolveSessionId home >>= \case
    Right sessionId -> TIO.putStr sessionId
    Left _ -> dieText "no session registered for this process. Run 'handler register' or start a new prompt first"
